packages/ramldoc-generator/scripts/download-rmf-codegen.js

```diff
@@ -0,0 +1,58 @@
+const fs = require('fs');
+const path = require('path');
+const fetch = require('node-fetch');
+const shelljs = require('shelljs');
+const hasbin = require('hasbin');
+const { rmfCodegenVersion } = require('../package.json');
+
+const binPath = path.join(__dirname, '../bin');
+const jarName = `rmf-codegen-${rmfCodegenVersion}.jar`;
+const jarPath = path.join(binPath, jarName);
+const downloadURI = `https://dl.bintray.com/vrapio/vrapio/io/vrap/rmf/codegen/cli-application/${rmfCodegenVersion}/cli-application-${rmfCodegenVersion}-all.jar`;
+
+const abortIfError = result => {
+  if (result.code > 0) {
+    throw new Error(result.stderr);
+  }
+};
+const downloadArchive = async url => {
+  // Download the archive
+  const res = await fetch(url);
+  const fileStream = fs.createWriteStream(jarPath);
+  await new Promise((resolve, reject) => {
+    res.body.pipe(fileStream);
+    res.body.on('error', error => {
+      reject(error);
+    });
+    fileStream.on('finish', resolve);
+  });
+
+  // Assign proper write permissions to the file
+  abortIfError(shelljs.chmod(755, jarPath));
+};
+
+if (!hasbin.sync('java')) {
```

Comment:
It's sufficient to do something like this:

```js
const isJavaInstalled = () => {
  try {
    const result = shelljs.exec('java -version', { silent: true });
    return result.code === 0;
  } catch (error) {
    return false;
  }
}
```

Let's avoid using unnecessary dependencies.